File: v1/src/main/java/com/api/v1/services/AllPaymentsDeletionServiceImpl.java
package com.api.v1.services;

import com.api.v1.domain.PaymentRepository;
import com.api.v1.exceptions.EmptyPaymentEntityException;
import com.api.v1.utils.PaymentFinderUtil;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
class AllPaymentsDeletionServiceImpl implements AllPaymentsDeletionService {

    @Autowired
    private PaymentFinderUtil paymentFinderUtil;

    @Autowired
    private PaymentRepository repository;

    @Override
    public Mono<Void> deleteAll() {
        return repository
                .findAll()
                .hasElements()
                .flatMap(hasElements -> {
                    if (!hasElements) return Mono.error(EmptyPaymentEntityException::new);
                    return repository.deleteAll();
                });
    }

    @Override
    public Mono<Void> deleteByOrderNumber(String orderNumber) {
        return paymentFinderUtil
                .find(orderNumber)
                .flatMap(payment -> repository.delete(payment));
    }

}

File: kotlin/v2/src/main/kotlin/com/api/v2/payment/services/AllPaymentsDeletionServiceImpl.kt
package com.api.v2.payment.services

import com.api.v2.payment.domain.PaymentRepository
import com.api.v2.payment.dtos.EmptyPaymentEntity
import com.api.v2.payment.utils.PaymentFinderUtil
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.count
import kotlinx.coroutines.withContext
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
private class AllPaymentsDeletionServiceImpl: AllPaymentsDeletionService {

    @Autowired
    lateinit var paymentFinderUtil: PaymentFinderUtil

    @Autowired
    lateinit var paymentRepository: PaymentRepository

    override suspend fun deleteAll() {
        withContext(Dispatchers.IO) {
            if (paymentRepository.findAll().count() == 0) {
                throw EmptyPaymentEntity()
            }
            paymentRepository.deleteAll()
        }
    }

    override suspend fun deleteByOrderNumber(orderNumber: String) {
        withContext(Dispatchers.IO) {
            val payment = paymentFinderUtil.find(orderNumber)
            paymentRepository.delete(payment)
        }
    }

}